community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveIntStack.java

```diff
@@ -62,7 +62,22 @@ public void close()
     @Override
     public PrimitiveIntIterator iterator()
     {
-        throw new UnsupportedOperationException( "Please implement" );
+        return new PrimitiveIntIterator()
+        {
+            int idx = 0;
+
+            @Override
+            public boolean hasNext()
+            {
+                return idx <= cursor;
+            }
+
+            @Override
+            public int next()
+            {
+                return array[idx++];
```

Comment:
Imho check like: `if( !hasNext() ) throw new NoSuchElementException();` is required